t/rose-loc-handlers/test_rsync_loc_handlers.t

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# Copyright (C) British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Tests behaviour of rose/loc_handlers/rsync_remote_check.py. Ideally this
+# should be run against as many versions of Python as possible.
+# See: https://github.com/metomi/rose/issues/2609
+
+. $(dirname $0)/test_header
+
+tests 4
+
+# Print the Python version to stderr, just to check.
+python --version >&2
+
+# Create a folder containing both files and directories.
+mkdir -p foo/baz
+echo "HI" > foo/bar
+echo "THERE" > foo/baz/qux
+chmod 752 foo/baz/qux
+chmod +x foo/bar
```

Comment:
You might want to use an absolute value here too (e.g. `755`) as `+x` just adds user execute (i.e. makes the first digit a `7`) but doesn't change the other fields.